### PR TITLE
Fixes for the new parser

### PR DESCRIPTION
**Pull request by HalidOdat**
Now the parser tests pass, but there are other tests that fail.

```
test result: FAILED. 92 passed; 35 failed; 1 ignored; 0 measured; 0 filtered out
```

The tests that fail are:
```
failures:
    builtins::array::tests::every
    builtins::array::tests::fill
    builtins::array::tests::filter
    builtins::array::tests::find
    builtins::array::tests::find_index
    builtins::array::tests::for_each
    builtins::array::tests::for_each_push_value
    builtins::array::tests::is_array
    builtins::array::tests::map
    builtins::boolean::tests::construct_and_call
    builtins::boolean::tests::constructor_gives_true_instance
    builtins::boolean::tests::instances_have_correct_proto_set
    builtins::function::tests::check_arguments_object
    builtins::number::tests::value_of
    builtins::regexp::tests::test_constructors
    builtins::regexp::tests::test_exec
    builtins::regexp::tests::test_last_index
    builtins::regexp::tests::test_to_string
    builtins::string::tests::concat
    builtins::string::tests::concat
    builtins::string::tests::construct_and_call
    builtins::string::tests::ends_with
    builtins::string::tests::match_all
    builtins::string::tests::repeat
    builtins::string::tests::replace_with_function
    builtins::string::tests::starts_with
    builtins::string::tests::test_match
    exec::tests::array_rest_with_arguments
    exec::tests::object_field_set
    exec::tests::spread_with_arguments
    exec::tests::test_early_return
    exec::tests::test_short_circuit_evaluation
    exec::tests::test_tilde_operator
    syntax::lexer::tests::check_line_numbers
    syntax::lexer::tests::check_single_line_comment
```

### CODE REVIEW

**Comment by HalidOdat** [comment]
Progress:
```
test result: FAILED. 94 passed; 33 failed; 1 ignored; 0 measured; 0 filtered out
```

File: boa/src/syntax/parser/mod.rs [hunk 1]
```diff
@@ -2244,10 +1968,10 @@ impl Parser {
             }
             TokenKind::Punctuator(Punctuator::OpenBracket) => self.read_array_literal(),
             TokenKind::Punctuator(Punctuator::OpenBlock) => self.read_object_literal(),
-            TokenKind::Identifier(ref i) if i == "true" => Ok(Node::Const(Const::Bool(true))),
-            TokenKind::Identifier(ref i) if i == "false" => Ok(Node::Const(Const::Bool(false))),
+            TokenKind::BooleanLiteral(boolean) => Ok(Node::Const(Const::Bool(boolean))),
+            // TODO: ADD TokenKind::UndefinedLiteral
```
Comment (jasonwilliams) [inline boa/src/syntax/parser/mod.rs]: Would be good to make this into a "good first" issue

File: boa/src/syntax/parser/mod.rs [hunk 2]
```diff
@@ -2364,6 +2088,24 @@ impl Parser {
             return Ok(PropertyDefinition::Property(to_string(tok.kind), val));
         }
 
+        // TODO: Slit into separate function: read_propery_method_definition
```
Comment (jasonwilliams) [inline boa/src/syntax/parser/mod.rs]: Same for this

**Comment by HalidOdat** [comment]
Now we are at:
```
test result: FAILED. 108 passed; 19 failed; 1 ignored; 0 measured; 0 filtered out
```
The rest of the errors are caused because `exec` does not implement some things like  `Node::New`.
I think we should merge this, so we can start on fixing the `exec` issues.

**Pull request merged**